builder/vmware/common/driver.go

```diff
@@ -257,28 +280,40 @@ func runAndLog(cmd *exec.Cmd) (string, string, error) {
 }
 
 // Still used for Workstation and Player until conversion.
-func normalizeVersion(version string) (string, error) {
-	i, err := strconv.Atoi(version)
-	if err != nil {
-		return "", fmt.Errorf("returned a non-integer version %q: %s", version, err)
+func normalizeVersion(version string) ([]int, error) {
```

Comment:
I wonder since we do comparisons after this, could we use a version library like `hashicorp/go-version`? That's assuming the version returned by vmware is semver-compatible, if not though feel free to disregard

Reply:
Pushed d5629c4 with an update that transitions to using `hashicorp/go-version` instead.

Did a quick test that had positive results.

```go

package main

import (
    "testing"
    "github.com/hashicorp/go-version"
)

func TestCompareVersions(t *testing.T) {
    tests := []struct {
        versionFound    string
        versionRequired string
        product         string
        expectError     bool
    }{
        {"1.0.0", "1.0.0", "ProductA", false},
        {"1.0.1", "1.0.0", "ProductA", false},
        {"1.0.0", "1.0.1", "ProductA", true},
        {"2.0.0", "1.0.0", "ProductA", false},
        {"1.0.0", "2.0.0", "ProductA", true},
    }

    for _, test := range tests {
        err := compareVersions(test.versionFound, test.versionRequired, test.product)
        if test.expectError && err == nil {
            t.Errorf("expected error but got none for versions %s and %s", test.versionFound, test.versionRequired)
        }
        if !test.expectError && err != nil {
            t.Errorf("did not expect error but got %v for versions %s and %s", err, test.versionFound, test.versionRequired)
        }
    }
}

func TestCompareVersionObjects(t *testing.T) {
    tests := []struct {
        versionFound    string
        versionRequired string
        product         string
        expectError     bool
    }{
        {"1.0.0", "1.0.0", "ProductA", false},
        {"1.0.1", "1.0.0", "ProductA", false},
        {"1.0.0", "1.0.1", "ProductA", true},
        {"2.0.0", "1.0.0", "ProductA", false},
        {"1.0.0", "2.0.0", "ProductA", true},
    }

    for _, test := range tests {
        vFound, err := version.NewVersion(test.versionFound)
        if err != nil {
            t.Fatalf("invalid version found: %v", err)
        }
        vRequired, err := version.NewVersion(test.versionRequired)
        if err != nil {
            t.Fatalf("invalid version required: %v", err)
        }

        err = compareVersionObjects(vFound, vRequired, test.product)
        if test.expectError && err == nil {
            t.Errorf("expected error but got none for versions %s and %s", test.versionFound, test.versionRequired)
        }
        if !test.expectError && err != nil {
            t.Errorf("did not expect error but got %v for versions %s and %s", err, test.versionFound, test.versionRequired)
        }
    }
}
```

```sh
Running tool: /opt/homebrew/bin/go test -timeout 30s -run ^TestCompareVersions$ github.com/hashicorp/packer-plugin-vmware/builder/vmware/common

ok  	github.com/hashicorp/packer-plugin-vmware/builder/vmware/common	(cached)
```